alacritty/src/renderer/mod.rs

```diff
@@ -218,3 +219,55 @@ impl Renderer {
         }
     }
 }
+
+struct GlExtensions {
+    extensions: Box<dyn Iterator<Item = Result<&'static str, Utf8Error>>>,
```

Comment:
A `Box<Iterator>` as storage on a struct seems somewhat unconventional. At that point why not just use a `Vec`?

Reply:
Because we don't collect? It's another iterator that handled 2 sources of extensions. it's purely for efficiency reasons, since we have 2 iterators and they both read static strings from some static storage, so having iterator like this won't allocate anything for elements used in the iterator.

By using `Vec` we'll have to collect and store everything, and there won't be a use for iterator at all. Though, we can likely refactor to some lazy initialized `HashSet<CStr>`.